apis/exoscale/v1/opensearch_types.go

```diff
@@ -0,0 +1,98 @@
+package v1
+
+import (
+	"reflect"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/crossplane/crossplane-runtime/pkg/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// OpenSearchParameters are the configurable fields of a OpenSearch instance.
+type OpenSearchParameters struct {
+	// OpenSearchSettings contains additional OpenSearch settings.
+	Maintenance        MaintenanceSpec `json:"maintenance,omitempty"`
+	Backup             BackupSpec      `json:"backup,omitempty"`
+	DBaaSParameters    `json:",inline"`
+	Zone               Zone                 `json:"zone"`
```

Comment:
Zone requirers kubebuilder validation annotation. Look at other examples - the comments with `// +kubebuilder` are significant for code generation